Assets/Script/Song/SongSources.cs

```diff
@@ -282,32 +285,25 @@ private static async UniTask DownloadSources(LoadingContext context)
 
         private static void ReadSources()
         {
+            // Read custom sources - they are read first so user replacements are prioritized
+            // Create the folder if it doesn't exist
+            Directory.CreateDirectory(CustomSourcesFolder);
+            // Read index.json if it exists; otherwise fallback to reading PNGs individually later on
+            string customIndexPath = Path.Combine(CustomSourcesFolder, "index.json");
+            var customSourcesRead = false;
```

Comment:
Nitpick: use explicit type for built-in types:

```suggestion
            bool customSourcesRead = false;
```